wear/src/main/java/com/eveningoutpost/dexdrip/Models/Bubble.java

```diff
@@ -0,0 +1,163 @@
+        final BridgeResponse reply = new BridgeResponse();
+        int first = 0xff & buffer[0];
+        if (first == 0x80) {
+            PersistentStore.setString("Bubblebattery", Integer.toString(buffer[4]));
+            Pref.setInt("bridge_battery", buffer[4]);
+            PersistentStore.setString("BubbleHArdware", Integer.toString(buffer[2]) + ".0");
+            PersistentStore.setString("BubbleFirmware", Integer.toString(buffer[1]) + ".0");
+            ByteBuffer ackMessage = ByteBuffer.allocate(6);
+            ackMessage.put(0, (byte) 0x02);
+            ackMessage.put(1, (byte) 0x01);
+            ackMessage.put(2, (byte) 0x00);
+            ackMessage.put(3, (byte) 0x00);
+            ackMessage.put(4, (byte) 0x00);
+            ackMessage.put(5, (byte) 0x2B);
+            reply.add(ackMessage);
+            s_full_data = null;
+            Log.e(TAG, "reset data");
+            return getBubbleResponse();
+        }
+        if (first == 0xC0) {
+            String SensorSn = LibreUtils.decodeSerialNumberKey(Arrays.copyOfRange(buffer, 2, 10));
+            PersistentStore.setString("LibreSN", SensorSn);
+            return reply;
+        }
+        if (first == 0x82) {
+            int expectedSize = lens + BUBBLE_FOOTER;
+            if (s_full_data == null) {
+                InitBuffer(expectedSize);
+            }
+            addData(buffer);
+            PersistentStore.setBoolean("CobbleNoSensor", false);
+            return reply;
+
+        }
+
+        if (first == 0xBF) {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+            Log.e(TAG, "No sensor has been found");
+            reply.setError_message(gs(R.string.no_sensor_found));
+            s_full_data = null;
+            errorCount++;
+            if (errorCount <= 2) {
+                return getBubbleResponse();
+            }
+            PersistentStore.setBoolean("CobbleNoSensor", true);
+            return reply;
+        }
+
+
+        return reply;
+    }
+
+
+    static void addData(byte[] buffer) {
+        System.arraycopy(buffer, 4, s_full_data, s_acumulatedSize, 16);
+        s_acumulatedSize = s_acumulatedSize + buffer.length - 4;
+        AreWeDone();
+    }
+
+
+    static void AreWeDone() {
+        if (s_acumulatedSize < lens) {
+            return;
+        }
+        long now = JoH.tsl();
+        String SensorSn = PersistentStore.getString("LibreSN");
+
+
+        byte[] data = Arrays.copyOfRange(s_full_data, 0, 344);
+        boolean checksum_ok = NFCReaderX.HandleGoodReading(SensorSn, data, now, true);
+        int expectedSize = lens + BUBBLE_FOOTER;
+        InitBuffer(expectedSize);
+        errorCount = 0;
+        Log.e(TAG, "We have all the data that we need " + s_acumulatedSize + " checksum_ok = " + checksum_ok + HexDump.dumpHexString(data));
+        if (!checksum_ok) {
+            int count = Pref.getInt("CobbleCheckFlase", 0);
```

Comment:
Please use `PersistentStore.setLong() and .getLong()` instead as this is not a preference.

Reply:
Fixed